### PR TITLE
[Go] Fix for 'Invalid code for files array in multipart/form-data request'…

### PR DESCRIPTION
**Pull request by aliakseiz**
Fixes invalid code generation, when multipart/form-data requestBody contains array of files (OpenAPITools#8093).

In addition to existing `ReadFormFileToTempFile` function a new one introduced `ReadFormFilesToTempFiles` returning a slice of `*os.File`. 

To avoid code duplication, common part related to `FileHeader` reading and storing in temporary file moved to new `readFileHeaderToTempFile` function.

`controller-api.mustache` updated to invoke corresponding function, depending on either single or multiple files are in a request.

Condition verifying the `os` package import extended in `GoServerCodegen.java` with a slice of files type.

Replaced a temporary directory path `"tmp"` with `""` in `ioutil.TempFile` call to make `TempFile()` using the default `os.TempDir` directory. 

### Testing

Existing automated tests passed. No new tests added.

Built project locally, manually tested generated code with different specifications (single file, array of files, various file formats).

### Technical committee
@antihax @grokify @kemokemo @bkabrda

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by wing328** [comment]
Please also run `./bin/utils/ensure-up-to-date` to update the samples so that CI can verify the change.

**Comment by aliakseiz** [comment]
> 
> 
> Please also run `./bin/utils/ensure-up-to-date` to update the samples so that CI can verify the change.

Thanks @wing328 !  
The `ensure-up-to-date` script updated quite a lot of files with only change of slash to backslash (executed on Windows). Not sure if that could cause any troubles for CI.

**Comment by aliakseiz** [comment]
It looks that _circleci_ build failed due to network connection issue:
```
[info] 	[SUCCESSFUL ] ch.qos.logback#logback-core;1.2.3!logback-core.jar (866ms)
[warn] 	[FAILED     ] net.sourceforge.htmlunit#htmlunit;2.27!htmlunit.jar: Connection reset (867ms)
[warn] 	[FAILED     ] com.typesafe.play#play-docs_2.12;2.6.18!play-docs_2.12.jar: Connection reset (606ms)
...
```

@wing328 can you please rerun the build?

**Comment by wing328** [comment]
@aliakseiz thanks for the PR. What about using tabs instead of 4-spaces?

**Comment by aliakseiz** [comment]
No problem, @wing328 .
Fixed.

**Review by wing328 — APPROVED** [review]
LGTM

File: modules/openapi-generator/src/main/resources/go-server/routers.mustache [hunk 1]
```diff
@@ -66,28 +66,61 @@ func EncodeJSONResponse(i interface{}, status *int, w http.ResponseWriter) error
...
+		}
+
+		files = append(files, file)
+	}
+
+	return files, nil
+}
+
+// readFileHeaderToTempFile reads multipart.FileHeader and writes it to a temporary file
+func readFileHeaderToTempFile(fileHeader *multipart.FileHeader) (*os.File, error) {
```
Comment (ys) [inline modules/openapi-generator/src/main/resources/go-server/routers.mustache:98]: multipart is not defined as an import here. We should add `"mime/multipart"` in the import list

**Pull request merged**